### PR TITLE
XL/erasure: refactor erasureReadFile. Move parallelRead into a separate function.

### PR DESCRIPTION
**Pull request by krishnasrinivas**
- getReadDisk was not a recursive function. Hence parallelRead as recursion looked odd and forced.
- move parallelRead() as a separate function so that it can be unit tested.

### CODE REVIEW

File: erasure-readfile.go [hunk 1]
```diff
@@ -125,6 +125,50 @@ func getReadDisks(orderedDisks []StorageAPI, index int, dataBlocks int) (readDis
 	return nil, 0, errXLReadQuorum
 }
 
+// parallelRead - reads chunks in parallel from the disks specified in []readDisks.
+func parallelRead(volume, path string, readDisks []StorageAPI, orderedDisks []StorageAPI, enBlocks [][]byte, blockOffset int64, curChunkSize int64, bitRotVerify func(diskIndex int) bool) {
```
Comment (harshavardhana) [inline erasure-readfile.go:129]: Can you explore the possibility of adding a unit test for this code path?

Comment (krishnasrinivas) [inline erasure-readfile.go:129]: will do

Comment (krishnasrinivas) [inline erasure-readfile.go:129]: @harshavardhana will add the unit test as part of https://github.com/minio/minio/issues/2010

**Comment by harshavardhana** [comment]
> @harshavardhana will add the unit test as part of #2010

Okay

**Pull request merged**